surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+>
+> We're using "Linux" here rather than grouping all UNIXes together, to allow
+> us to gauge interest in specific other UNIXes via the fill-in-the-blank
+> "other" option. If we grouped UNIXes together, users of other UNIX systems
+> wouldn't be visible; let's try to capture the level of interest in those
+> systems. As with many questions with an open "other" response; if any
+> specific answer appears frequently, we can add it to future surveys to reduce
+> the amount of work needed to process responses.
+
+### Which operating systems or runtimes do you develop Rust software for?
+
+**Note**: this is specifically about which operating system or runtime you **target** not which system you use
+for development nor which specific architectures (e.g., x86 vs ARM) you target.
+
+Type: select all that apply (optional)
+
+- Linux (desktop or server)
+- Windows 10/11
+- Windows 8 or older
+- Mac OS
+- iOS
+- Android
+- Embedded platforms (with an operating system)
+- Embedded platforms (bare metal)
+- WebAssembly
+- Explicitly platform-independent (e.g., a library which does not interact with the operating system)
+- Other (open response)
+
+> **justification**
+>
+> This question can be used to figure out roughly what systems are being targeted as well as 
+> what OS stack is being developed against (i.e., desktop/server OS, mobile OS, embedded OS, bare metal)
+>
+> We're using "Linux" here rather than "*nix" or similar, with the same
+> justification as in the "Which operating systems do you use" question.
+>
+> We specifically care about the runtime environment being targeted. ISA and other machine specifics are
+> not what matters.
+
+### Which version(s) of Rust do you use for development (excluding CI and other automated testing)?
+
+Type: select all that apply (optional)
+
+- Current stable version
+- Previous stable version
+- A specific version of stable Rust equal to or newer than 1.67
+- A specific version of stable Rust older than 1.67
+- Beta release
+- Latest nightly
+- A specific version of nightly
+- Custom fork
+- I don't know
+- Other
+
+> **justification**
+>
+> Together with the following question, we can better determine what the spread of 
+> version usage is across the community.
+> We ask specifically about version 1.67 since it is, at the time of the survey, the 
+> version that was released ~1 year prior. Additionally, at the time of this writing
+> all major Linux distros have a version equal to or newer than this version.
+
+### If you use nightly for any tasks, why?
+
+Type: select all that apply (optional)
+
+- I don't use nightly
+- Out of habit
+- For a particular language feature or set of language features I need
+- I like to have access to all the latest features
+- To help test the nightly version for bugs
+- To provide design feedback on nightly features
+- For testing in CI
+- A crate dependency I use requires it
+- A tool I use requires it
+- Other (open response)
```

Comment:
Faster compile times is a common one I've seen.

Reply:
Mhmm :) Although perceptions and reality can diverge! See https://bevyengine.org/learn/book/getting-started/setup/#enable-fast-compiles-optional, where Bevy recommends it.

Reply:
Basically what I was commenting was that I'm personally not aware of nightly having "access to the latest performance improvements and "unstable" optimizations", apart from the time diff between nightly and stable, of course. I don't think that any performance optimizations are enabled by default in nightly that would be disabled in stable. So the bevy guide might be a bit misleading, unless I'm missing something. But that's not really relevant to this survey.

If we add the option, we could find out how many people share this perception. Hopefully it won't cause the readers to think that this is true and then migrate to nightly because of that (although it sure would be nice for increased testing :laughing:).

Anyway, that's too much overthinking for a simple change. I added a commit with the suggested answer.